### PR TITLE
feat(model_zoo): add llama2 fsdp finetuning example

### PR DESCRIPTION
**Pull request by Major-333**
### What changes were proposed in this pull request?

1. Added a model example for fine-tuning Llama2
2. Added a training image for building Llama2 fine-tuning
3. Added a job configuration example for Llama2 fine-tuning phase 2.

### Why are the changes needed?

Added a model example of fine-tuning the Llama2 model based on the FSDP training strategy.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Manual

### CODE REVIEW

**Comment by Major-333** [comment]
Documentation work is in progress.

File: model_zoo/pytorch/llama/configs/fsdp.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class fsdp_config:
+    mixed_precision: bool=True
+    use_fp16: bool=True
+    sharding_strategy: ShardingStrategy = ShardingStrategy.FULL_SHARD
+    checkpoint_type: StateDictType = StateDictType.SHARDED_STATE_DICT  # alternatively can use SHARDED_STATE_DICT save one file per rank, and can resize the world-size.
+    fsdp_activation_checkpointing: bool=True
+    pure_bf16: bool = True
+    optimizer: str= "AdamW"
+    
+
```
Comment (merlintang) [inline model_zoo/pytorch/llama/configs/fsdp.py:19]: dont make blank line here

File: model_zoo/pytorch/llama/configs/fsdp.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@dataclass
+class fsdp_config:
+    mixed_precision: bool=True
+    use_fp16: bool=True
+    sharding_strategy: ShardingStrategy = ShardingStrategy.FULL_SHARD
+    checkpoint_type: StateDictType = StateDictType.SHARDED_STATE_DICT  # alternatively can use SHARDED_STATE_DICT save one file per rank, and can resize the world-size.
+    fsdp_activation_checkpointing: bool=True
+    pure_bf16: bool = True
+    optimizer: str= "AdamW"
```
Comment (merlintang) [inline model_zoo/pytorch/llama/configs/fsdp.py:17]: does this configurable?

File: model_zoo/pytorch/llama/configs/peft.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (merlintang) [inline model_zoo/pytorch/llama/configs/peft.py:1]: does this ok?

**Review by merlintang — CHANGES_REQUESTED** [review]
can you split this pr into different parts? this would help us to review it.

**Comment by Major-333** [comment]
> 



> can you split this pr into different parts? this would help us to review it.

I'm really sorry. I didn't set up Github notifications to go to my primary email, so I missed many of your messages. I'll be closing this PR soon and will upload the changes in separate parts as per your request.

**Pull request closed without merging**